### PR TITLE
change JobAlert model, add test for alert date calculation

### LINKED ISSUE #25: Create job alert models
Create job alert feature and models

### PR DESCRIPTION
**Pull request by advapi**
* change JobAlert model: remove unnecessary field and change fields for further activity 
* add test for alert date calculation function
fixes: #25 , #63

### CODE REVIEW

File: joboard/settings.py [hunk 1]
```diff
@@ -124,3 +124,10 @@
 STATIC_URL = '/static/'
 LOGIN_REDIRECT_URL = '/'
 LOGOUT_REDIRECT_URL = '/'
+
+EMAIL_BACKEND = 'django.core.mail.backends.smtp.EmailBackend'
+EMAIL_HOST = 'smtp.gmail.com'
+EMAIL_USE_TLS = True
+EMAIL_PORT = 587
+EMAIL_HOST_USER = 'JoBoardT@gmail.com'
+EMAIL_HOST_PASSWORD = 'Jb123456'
```
Comment (bardielle) [inline joboard/settings.py]: If it was a real application you can't write the password like that (visible for everyone).... 
So - remove it and save it in the DB

File: user_notification/migrations/0002_auto_20201223_2252.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import datetime
+from django.db import migrations, models
+
```
Comment (bardielle) [inline user_notification/migrations/0002_auto_20201223_2252.py]: Is it autogenreated file? The comment of "auto generated" is missing

File: user_notification/tests.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+import pytest
+from datetime import datetime
+
+
+# test for calc_check_day
+@pytest.mark.parametrize('n, x, expected', [
+    ("2020-12-27", 2, "2020-12-29"),
+    ("2020-12-27", 3, "2020-12-30"),
+])
+def testCalc_check_date(n, x, expected):
```
Comment (bardielle) [inline user_notification/tests.py]: testCalc_check_date -> test_calc_check_date

File: user_notification/models.py [hunk 4]
```diff
@@ -8,16 +10,25 @@ class JobAlert(models.Model):
...
+    job_alert_company_name = models.CharField(
+        max_length=20, blank=True, null=True)
 
     def __str__(self):
         return str(self.user_account_id)
 
+    # Calculate the date of the next alert
+    @classmethod
+    def calc_check_date(cls, last_check_date, frequency_in_days):
+        temp = datetime.strptime(last_check_date, "%Y-%m-%d").date()
```
Comment (bardielle) [inline user_notification/models.py]: what will happen if frequency_in_days would be a negative number?

File: user_notification/tests.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    assert JobAlert.calc_check_date(
+        n, x) == datetime.strptime(expected, "%Y-%m-%d").date()
+
+
+@pytest.mark.parametrize('n, x, expected', [
+    ("2020-12-27", 2, "2020-12-30"),
+    ("2020-12-27", 3, "2020-12-29"),
+])
+def testCalc_check_date2(n, x, expected):
```
Comment (bardielle) [inline user_notification/tests.py]: What is the different between those two tests?

Comment (advapi) [reply user_notification/tests.py]: just wanted to make sure the calculation really works by testing wrong answers too
should I delete the second test?

File: user_notification/models.py (hunk 4 above)
Comment (advapi) [reply user_notification/models.py]: I'll change the model field so it will set as PositiveIntegerField

Comment (bardielle) [reply user_notification/models.py]: :+1:

Comment (bardielle) [reply user_notification/models.py]: Also change the parameter name "temp" to be more meaningful

File: user_notification/tests.py (hunk 5 above)
Comment (bardielle) [reply user_notification/tests.py]: Yes... but maybe add a test of failure cases (for example negative numbers, invalid date, etc...

Comment (advapi) [reply user_notification/tests.py]: something like that:
with pytest.raises(ValueError, ...) ?

File: user_notification/migrations/0002_auto_20201223_2252.py (hunk 2 above)
Comment (advapi) [reply user_notification/migrations/0002_auto_20201223_2252.py]: fixed

File: user_notification/tests.py (hunk 3 above)
Comment (advapi) [reply user_notification/tests.py]: fixed

File: user_notification/models.py (hunk 4 above)
Comment (advapi) [reply user_notification/models.py]: fixed

File: user_notification/tests.py (hunk 5 above)
Comment (advapi) [reply user_notification/tests.py]: is it possible to approve this test for now and I'll try to learn how to make a test of failure cases? 
I've deleted the 2nd test that was unnecessary

File: joboard/settings.py (hunk 1 above)
Comment (advapi) [reply joboard/settings.py]: deleted this for now. I will upload that in a separate PR

File: user_notification/tests.py (hunk 5 above)
Comment (bardielle) [reply user_notification/tests.py]: Better remove that test...

Comment (bardielle) [reply user_notification/tests.py]: I saw you removed it...

File: user_notification/models.py [hunk 6]
```diff
@@ -8,16 +11,30 @@ class JobAlert(models.Model):
...
+    job_alert_company_name = models.CharField(
+        max_length=20, blank=True, null=True)
 
     def __str__(self):
         return str(self.user_account_id)
 
+    # Calculate the date of the next alert
+    @classmethod
+    def calc_check_date(cls, last_check_date, frequency_in_days):
+        if isinstance(last_check_date, datetime):
```
Comment (bardielle) [inline user_notification/models.py]: still - if frequency_in_days is a negative number would will be the output? 
You don't want to pass the function - need to raise an exception

Comment (advapi) [reply user_notification/models.py]: fixed

**Review by bardielle — APPROVED** [review]
/lgtm
